MapboxCoreNavigation/RouteController.swift

```diff
@@ -805,15 +816,9 @@ extension RouteController: CLLocationManagerDelegate {
             
             let lastKnownUserAbsoluteDistance = routeProgress.currentLegProgress.currentStepProgress.userDistanceToManeuverLocation
             
-            // The objective here is to make sure the user is moving away from the maneuver location
-            // This helps on maneuvers where the difference between the exit and enter heading are similar
-            if  userAbsoluteDistance <= lastKnownUserAbsoluteDistance! {
-                routeProgress.currentLegProgress.currentStepProgress.userDistanceToManeuverLocation = userAbsoluteDistance
-            }
-            
             if routeProgress.currentLegProgress.upComingStep?.maneuverType == ManeuverType.arrive {
                 alertLevel = .arrive
-            } else if courseMatchesManeuverFinalHeading {
+            } else if courseMatchesManeuverFinalHeading || (userAbsoluteDistance > lastKnownUserAbsoluteDistance! && lastKnownUserAbsoluteDistance! > RouteControllerManeuverZoneRadius) {
```

Comment:
Are we guaranteed to have a `lastKnownUserAbsoluteDistance` here? Let’s move the `!`s up to where we define `lastKnownUserAbsoluteDistance` and consider adding a `guard` or `if` statement to handle the case where we don’t have one.